src/base/utils/fs.h

```diff
@@ -67,6 +67,40 @@ namespace Utils
         /* End of Qt4 code */
 
         QString cacheLocation();
+
+        /// Utility class to defer file deletion
+        class FileProcessingGuard
+        {
+        public:
+            FileProcessingGuard(const QString &path = QString());
+            ~FileProcessingGuard();
+
+            /// Cancels deferred file deletion
+            void releaseFile();
+
+        private:
+            QString m_path;
+        };
+
+        /// reads settings for .torrent files from preferences
+        /// and sets up file guard accordantly
+        class TorrentFileProcessignGuard : private FileProcessingGuard
```

Comment:
It is better to consider this class as a separate component and put it in its own *.h/*.cpp files (along with its base class or merge these two classes into one). Utils must contain only simple logic routines that are independent from the logic of the application.

P.S. You do not need to rush in search of that does not meet my approval. Perhaps you will find it. As I have repeatedly said, we are constantly improving our code, but we can not change it all at once. Therefore, there is still contains a lot of "bad" fragments which can not be regarded as examples to follow.